src/store/actions/userSession.actions.js

```diff
@@ -115,20 +144,25 @@ export const logout = () => dispatch => {
 export const getUserStatus = () => dispatch => {
   dispatch(checkLoggedStatus())
   const JWTtoken = getAccessToken()
+  const username = getUserNameFromToken(JWTtoken) || getUserName()
+  const userrole = getUserRoleFromToken(JWTtoken) || true
 
   if (JWTtoken) {
     dispatch(loginSuccess(
-      getUserName(JWTtoken),
-      getUserAdminRole(JWTtoken)
+      username,
+      userrole
     ))
   } else {
     history.push('/login')
   }
 }
 
-function getUserName (JWTtoken) {
-  return jwt.decode(JWTtoken).sub
+function getUserNameFromToken (token) {
+  const payload = jwt.decode(token)
```

Comment:
Let's have declarations a bit separated from the rest of the functionality (via extra line).